### PR TITLE
[WIP] Add obal-fmt

### PR DESCRIPTION
**Pull request by ekohl**
This is a very simple tool that formats the YAML manifest and can be used to enforce a style.

### CODE REVIEW

**Comment by ehelms** [comment]
I am commenting on the usage in f-packaging as I think its easier to see there.

**Comment by ekohl** [comment]
> I am commenting on the usage in f-packaging as I think its easier to see there.

That was exactly the reason why I opened that PR.

File: obal/__init__.py [hunk 1]
```diff
@@ -7,6 +7,7 @@
 import os
 import sys
 
+import yaml
```
Comment (evgeni) [inline obal/__init__.py]: Strictly speaking, this needs `PyYAML` as a dep in `setup.py` etc, right? (it gets pulled in by `ansible` anyways, but let's be explicit)

File: obal/__init__.py [hunk 2]
```diff
@@ -105,6 +106,33 @@ def generate_ansible_args(inventory_path, playbook_path, args):
...
+        argcomplete.autocomplete(parser)
+
+    args = parser.parse_args(cliargs)
+
+    if not os.path.exists(args.manifest):
+        print("Could not find your package_manifest.yaml")
+        exit(1)
+
+    with open(args.manifest) as manifest_fp:
+        manifest = yaml.load(manifest_fp)
```
Comment (evgeni) [inline obal/__init__.py]: should that be `safe_load`?

**Comment by ekohl** [comment]
Updated to maintain ordering when ```packages``` is a dictionary key. I would like this to be more explicit rather than respecting current ordering.

**Comment by ekohl** [comment]
I want to look into compatibility with ruamel 0.13 since that's in Fedora 28 and ruamel 0.15 since that's on PyPi and in F29.

**Comment by evgeni** [comment]
closed in favor of the obsah issue

**Pull request closed without merging**